File: Selenium/Project/Activity5.java
package project;

import org.testng.annotations.Test;

import io.github.bonigarcia.wdm.WebDriverManager;

import org.testng.annotations.BeforeClass;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.ui.Select;
import org.testng.annotations.AfterClass;

public class Activity5 {
	
	WebDriver driver;
	
	  @BeforeClass
	  public void beforeClass() {
		  
		  WebDriverManager.firefoxdriver().setup();
		  driver = new FirefoxDriver();
		  driver.get("http://alchemy.hguy.co/orangehrm");
	  }
	  
  @Test
  public void editUserInfo() {
	  
	  	WebElement username = driver.findElement(By.id("txtUsername"));
		WebElement password = driver.findElement(By.id("txtPassword"));
		username.sendKeys("orange");
		password.sendKeys("orangepassword123");
		WebElement loginButton = driver.findElement(By.name("Submit"));
		loginButton.click();
		driver.findElement(By.id("menu_pim_viewMyDetails")).click();
		driver.findElement(By.id("btnSave")).click();
		WebElement fullName = driver.findElement(By.id("personal_txtEmpFirstName"));
		WebElement lastName = driver.findElement(By.name("personal[txtEmpLastName]"));
		WebElement gender1 = driver.findElement(By.id("personal_optGender_1"));
		WebElement gender2 = driver.findElement(By.id("personal_optGender_2"));
		Select nationality = new Select(driver.findElement(By.name("personal[cmbNation]")));
		WebElement doB = driver.findElement(By.id("personal_DOB"));
		fullName.clear();
		lastName.clear();
		doB.clear();
		fullName.sendKeys("Peter");
		gender1.click();
		nationality.selectByVisibleText("Indian");
		doB.sendKeys("1997-10-05");
		WebElement saveButton = driver.findElement(By.id("btnSave"));
		saveButton.click();
		System.out.println("URL after saving the details is : " + driver.getCurrentUrl());
  }
  

  @AfterClass
  public void afterClass() {
	  
	  driver.close();
	  
  }

}

File: Selenium/Project/activity5.py
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.common.action_chains import ActionChains
from selenium.webdriver.support.ui import Select

service = FirefoxService(GeckoDriverManager().install())

with webdriver.Firefox(service=service) as driver :
    Action = ActionChains(driver)
    driver.get("http://alchemy.hguy.co/orangehrm")

    username = driver.find_element(By.ID,"txtUsername")
    password = driver.find_element(By.ID,"txtPassword")

    username.send_keys("orange")
    password.send_keys("orangepassword123")

    loginButton = driver.find_element(By.NAME,"Submit")
    loginButton.click()

myInfo = driver.find_element(By.ID,"menu_pim_viewMyDetails")
myInfo.click()
editButton = driver.find_element(By.ID,"btnSave")
editButton.click()
		
fullName = driver.find_element(By.ID , "personal_txtEmpFirstName")
lastName = driver.find_element(By.NAME, "personal[txtEmpLastName]")
gender1 = driver.find_element(By.ID, "personal_optGender_1")
gender2 = driver.find_element(By.ID, "personal_optGender_2")
select = Select(driver.find_element_by_name('personal[cmbNation]'))
doB = driver.find_element(By.ID,"personal_DOB")
		
fullName.clear()
lastName.clear()
doB.clear()
fullName.sendKeys("Peter")
lastName.sendKeys("KT")
gender1.click()
select.selectByVisibleText("Indian")
doB.sendKeys("1997-10-05")
saveButton = driver.find_element(By.ID,"btnSave")
saveButton.click()
print("URL after saving the details is : " + driver.getCurrentUrl())